src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -437,6 +445,13 @@ ov::npuw::LLMCompiledModel::LLMCompiledModel(const std::shared_ptr<ov::Model>& m
 
     m_cfg.update(any_copy(npuw_llm_props));
 
+    if (m_name.find(NPUW_NAME_IDENTIFIER) != m_name.npos) {
+        LOG_DEBUG("LLMCompiledModel is being deserialized, skipping the full constructor flow...");
+        // Drop serialized from the name
+        m_name = m_name.substr(std::string(NPUW_NAME_IDENTIFIER).size());
+        return;
+    }
```

Comment:
Why again you do this

Reply:
In constructor of CompiledModel and LLMCompiledModel I somehow need to identify that it's being called from deserialize(). So I need some mechanism to distinguish default constructor behavior from a dummy deserialized one

Reply:
Done